src/flow/flow_graph.cpp

```
@@ -0,0 +1,341 @@
+            void Error(std::string message) {
+                errors.push_back({ layerId, transitionId, std::move(message) });
+            }
+        };
+
+        TriggerSpec ParseTrigger(nlohmann::json const& j, ParseContext& ctx) {
+            TriggerSpec spec;
+            if (!j.is_object()) {
+                ctx.Error("trigger must be an object");
+                return spec;
+            }
+            std::string typeStr = j.value("type", "");
```

Clang (14.0.6)

[error] variable 'typeStr' is not initialized

(cppcoreguidelines-init-variables,-warnings-as-errors)